modules/stochastic_tools/src/utils/GaussianProcessHandler.C

```diff
@@ -361,11 +369,16 @@ GaussianProcessHandler::getLossAdam(RealEigenMatrix & inputs, RealEigenMatrix &
   log_likelihood += -std::log(_K.determinant());
   log_likelihood -= _batch_size * std::log(2 * M_PI);
   log_likelihood = -log_likelihood / 2;
+  for (unsigned int ii = 0; ii < _num_tunable; ++ii)
+    log_likelihood += -std::log(p_std) - std::log(2.0 * M_PI) / 2.0 -
+                      Utility::pow<2>((std::log(theta(ii)) / p_std)) / 2.0;
```

Comment:
Are there other types of regularization that could be implemented in the future? If so, I'm thinking we could have a function that adds the regularization: something like `getRegularization(theta)`. This would have the logic to switch between regularization types.

Reply:
You're right. There are other types of regularizations. But all methods don't seem to work in the same way in terms of modifying the loss function. But for the Bayesian regularization, at least, I will generalize this by adding the `getRegularization()` function.

Reply:
These two are useful resources on priors for the GP hyper-params: https://mc-stan.org/docs/2_22/stan-users-guide/fit-gp-section.html 

https://www.tensorflow.org/probability/examples/Gaussian_Process_Regression_In_TFP